ambari-server/src/main/java/org/apache/ambari/server/api/services/ClusterService.java

```diff
@@ -860,4 +879,81 @@ ResourceInstance createArtifactResource(String clusterName, String artifactName)
 
     return createResource(Resource.Type.Artifact, mapIds);
   }
+
+  private boolean parseBody(String body) {
+    JsonObject jsonObject = gson.fromJson(body, JsonObject.class);
```

Comment:
suggest handleException  
 try {
          jsonObject = gson.fromJson(body, JsonObject.class);
      } catch (JsonSyntaxException e) {
          return false;
      }

Reply:
The exception is managed in the method from which the call originates.